backend/support/cog_over.py

```diff
@@ -0,0 +1,213 @@
+from collections import defaultdict
+from audit.models import SingleAuditChecklist
+from .models import CognizantBaseline, CognizantAssignment
+from dissemination.models import General
+from dissemination.hist_models.census_2019 import CensusGen19, CensusCfda19
+from dissemination.hist_models.census_2022 import CensusGen22
+from django.db.models.functions import Cast
+from django.db.models import BigIntegerField, Q
+# import logging
+
+# logger = logging.getLogger(__name__)
+
+
+COG_LIMIT = 50_000_000
+DA_THRESHOLD_FACTOR = 0.25
+
+
+def compute_cog_over(sac: SingleAuditChecklist):
+    """
+    Compute cog or oversight agency for the sac.
+    Return tuple (cog_agency, oversight_agency)
+    """
+    if not sac.federal_awards:
+        print(
```

Comment:
we can change this to `logger.warn` so that this will show up in new relic if it's ever hit

Reply:
Done